bigtable-client-core-parent/bigtable-hbase-integration-tests-common/src/test/java/com/google/cloud/bigtable/hbase/AbstractTestFilters.java

```diff
@@ -1547,6 +1547,52 @@ public void testSingleColumnValueFilter() throws IOException {
     Assert.assertEquals(0, results.length);
   }
 
+  @Test
+  public void testSingleColumnValueFilterNotEqualEmptyString() throws IOException {
+    // Set up:
+    // Row 1: f:qualifier1 = value1
+    // Row 2: f:qualifier1 = ""
+
+    // Cases to test:
+    // a: Filter NOT_EQUAL to EMPTY_STRING
+
+    byte[] rowKey1 = dataHelper.randomData("scvfrk1");
+    byte[] rowKey2 = dataHelper.randomData("scvfrk2");
+    byte[] qualifier1 = dataHelper.randomData("scvfq1");
+    byte[] nonEmptyStringValue = dataHelper.randomData("val1.1");
+    byte[] emptyStringValue = "".getBytes();
+
+
+    Table table = getDefaultTable();
+    Put put = new Put(rowKey1);
+    put.addColumn(COLUMN_FAMILY, qualifier1, nonEmptyStringValue);
+    table.put(put);
+
+    put = new Put(rowKey2);
+    put.addColumn(COLUMN_FAMILY, qualifier1, emptyStringValue);
```

Comment:
Empty value == null for Cloud Bigtable.  For rowKey2, I would think that `table.get(new Get(rowKey2))` will never return any values.

Reply:
Please add a second qualifier on both puts to make sure that we don't return rowKey2

Reply:
Table.get(new Get(rowKey2)) does return the row. 

Non empty value for second qualifier will return the rowKey2 if we put filter of NOT_EQUAL to empty string in cell value.

Reply:
I just tried adding second qualifier in both row with some string bytes in value, it returns both row with cells which has value(filtering out cell with empty value), using SingleColumnFIlter with NOT_EQUAL and empty string/value.